vault/request_forwarding.go

```diff
@@ -350,7 +350,7 @@ func (c *Core) ForwardRequest(req *http.Request) (int, http.Header, []byte, erro
 		req.URL.Path = origPath
 	}()
 
-	req.URL.Path = req.Context().Value(logical.CtxKeyOriginalRequestPath{}).(string)
+	req.URL.Path, _ = logical.ContextOriginalRequestPathValue(req.Context())
```

Comment:
❓ Would it make sense to check the `ok` return value here? Would there ever be a case that we would try to forward a request despite the request path value being empty?

Reply:
The previous code did not check if the value could be successfully cast to string, which is not great. I could introduce a check for ok and return an error if it's false. If it's the case that when the original request path doesn't get stored correctly in the Context the request forwarding doesn't stand a chance, then returning an error seems sensible.

Reply:
Ah, that I'm not necessarily that worried about it. As you said, it is going to be an error either way. As written, it would just result further down-stream.

Reply:
Actually, I wasn't too sure about that and just ran a test (`vault.TestCluster_ForwardRequests`), which I knew would run through the above code, through the debugger. After stepping over line 353, I reset the value of req.URL.Path to `""` and let the remainder of the test run but no error was returned, which tells me that this is a breeding ground for bugs.